WooCommerce/Classes/ViewRelated/AppCoordinator.swift

```diff
@@ -45,7 +46,8 @@ final class AppCoordinator {
          loggedOutAppSettings: LoggedOutAppSettingsProtocol = LoggedOutAppSettings(userDefaults: .standard),
          pushNotesManager: PushNotesManager = ServiceLocator.pushNotesManager,
          featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService,
-         purchasesManager: InAppPurchasesForWPComPlansProtocol? = nil) {
+         purchasesManager: InAppPurchasesForWPComPlansProtocol? = nil,
+         inAppPurchasesPlanManager: InAppPurchasesForWPComPlansProtocol = InAppPurchasesForWPComPlansManager()) {
```

Comment:
You can probably just use the one here?
```suggestion
         purchasesManager: InAppPurchasesForWPComPlansProtocol = InAppPurchasesForWPComPlansManager()) {
```

Though this may have a knock-on effect in the StoreCreationCoordinator which has this as nil, and so uses the WebPlanPurchase thing... I don't know the store creation flow at all though.

Reply:
Great catch.

I didn't even realize that we already have an instance of `InAppPurchasesForWPComPlansManager`. 🤦 

But I decided to have a separate one for Free trial check purposes to avoid having an undesired effect on the store creation flow. Done in https://github.com/woocommerce/woocommerce-ios/pull/10133/commits/371ae2b582972d394f39e3b712470652d366ed11

Reply:
@jaclync 👋 

I have injected another object into `AppCoordinator` confirming `InAppPurchasesForWPComPlansProtocol`. This is used to check for Free trial upgrade eligibility using IAP. 

I didn't alter and use the existing `purchasesManager` because it is injected into `StoreCreationCoordinator`, and I didn't want to have undesired effects on the store creation flow this close to the code freeze.

Let me know if you think this should be changed. I can address that next week. Thank you!